spec/controllers/media_objects_controller_spec.rb

```diff
@@ -417,6 +417,19 @@
     end
   end
 
+  describe "#save" do
+    it 'removes bookmarks that are no longer viewable' do
+      media_object = FactoryGirl.create(:published_media_object)
+      user = FactoryGirl.create(:public)
+      bookmark = Bookmark.create(document_id: media_object.pid, user_id: user.id) 
+      expect(Bookmark.exists? bookmark.id).to be_truthy
+      login_user media_object.collection.managers.first
+      request.env["HTTP_REFERER"] = '/'
+      get 'update_status', id: media_object.pid, status: 'unpublish'
+      expect(Bookmark.exists? bookmark.id).to be_falsey
```

Comment:
Looks like this works and would simplify the expects:

      expect {
        get 'update_status', id: media_object.pid, status: 'unpublish'
      }.to change { Bookmark.exists? bookmark }